### PR TITLE
*: add a flag to restrict cartesian product.

### PR DESCRIPTION
**Pull request by hanfei1991**
As the title, cartesian product is condisered very dangerous. We forbid it defaultly.
PTAL
@shenli @coocood @zimulala @tiancaiamao

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: executor/executor_test.go [hunk 1]
```diff
@@ -599,6 +600,15 @@ func (s *testSuite) TestJoin(c *C) {
 	result = tk.MustQuery("select a.c1 from t a , t1 b where a.c1 = b.c1 order by a.c1;")
 	result.Check(testkit.Rows("1", "2", "3", "4", "5", "6", "7"))
 
+	plan.AllowCartesianProduct = false
+	_, err := tk.Exec("select * from t, t1")
+	c.Check(err, NotNil)
```
Comment (zimulala) [inline executor/executor_test.go]: Check the error.

**Comment by coocood** [comment]
LGTM

File: tidb-server/main.go [hunk 2]
```diff
@@ -92,6 +93,9 @@ func main() {
 	if joinCon != nil && *joinCon > 0 {
 		plan.JoinConcurrency = *joinCon
 	}
+	if crossJoin != nil {
```
Comment (zimulala) [inline tidb-server/main.go]: Remove the useless judgement.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**